### PR TITLE
dockerfile: convert to Alpine base image

### PR DESCRIPTION
**Pull request by jzelinskie**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,26 +12,25 @@
...
-
-RUN go get -u github.com/cloudflare/cfssl/cmd/cfssl
-RUN go get -u github.com/cloudflare/cfssl/cmd/cfssljson
-RUN go get -u github.com/coreos/jwtproxy/cmd/jwtproxy # 06SEP2016
-RUN go get -u github.com/coreos/clair/cmd/clair
-RUN cd /go/src/github.com/coreos/clair && git checkout v1.2.6 && go install github.com/coreos/clair/cmd/clair
+RUN apk add --no-cache git bzr rpm xz supervisor gcc musl-dev && \
+    go get -u github.com/cloudflare/cfssl/cmd/cfssl && \
+    go get -u github.com/cloudflare/cfssl/cmd/cfssljson && \
+    apk del gcc && \
```
Comment (Quentin-M) [inline Dockerfile]: can `musl-dev` be removed?

Comment (jzelinskie) [inline Dockerfile]: I'm not sure, but I'd lean towards no -- it's required to get stdlib.h in the cgo used for cfssl, so it's definitely still being linked. I could probably uninstall it along with cfssl after we've generated the cert though. Either way, I don't think it'll save much space.

File: Dockerfile [hunk 2]
```diff
@@ -12,26 +12,25 @@
...
+    go get -u github.com/coreos/jwtproxy/cmd/jwtproxy && \
+    go get -u github.com/coreos/clair/cmd/clair && \
```
Comment (Quentin-M) [inline Dockerfile]: you can -d if you run `go install` afterwards, so we don't compile two versions.

File: Dockerfile [hunk 3]
```diff
@@ -12,26 +12,25 @@
...
+    cd /go/src/github.com/coreos/clair && \
+    git checkout v2.0.0-pre.4 && \
```
Comment (Quentin-M) [inline Dockerfile]: sad face

**Comment by Quentin-M** [comment]
RIPE CI OKR.

**Pull request merged**